backend/prisma/schema.prisma

```diff
@@ -0,0 +1,69 @@
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+// Looking for ways to speed up your queries, or scale easily with your serverless or edge functions?
+// Try Prisma Accelerate: https://pris.ly/cli/accelerate-init
+
+generator client {
+  provider = "prisma-client-js"
+  output   = "../generated/prisma"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model User {
+  id                    Int             @id @default(autoincrement())
+  name                  String
+  username              String          @unique
+  password              String
+  role                  Role
+  servicesOffered       String?
+  availabilities        Availibility[]
+  clientAppointments    Appointment[]   @relation("ClientAppointments")
+  providerAppointments  Appointment[]   @relation("ProviderAppointments")
+  reviewsWritten        Review[]        @relation("ClientReviews")
+  reviewsRecieved       Review[]        @relation("ProviderReviews")
+}
+
+model Availibility {
+  id                    Int             @id @default(autoincrement())
+  userId                Int 
+  user                  User            @relation(fields: [userId], references: [id])            
+  startDateTime         DateTime
+  endDateTime           DateTime
+}
+
+model Appointment {
+  id                    Int             @id @default(autoincrement())
+  clientId              Int  
+  client                User            @relation("ClientAppointments", fields: [clientId], references: [id]) 
+  providerId            Int 
+  provider              User            @relation("ProviderAppointments", fields: [providerId], references: [id])
+  dateTime              DateTime
+  status                AppointmentStatus
+  serviceType           String
+  notes                 String?
+}
+
+model Review {
+  id                    Int             @id @default(autoincrement())
+  clientId              Int
+  client                User            @relation("ClientReviews", fields: [clientId], references: [id])
+  providerId            Int
+  provider              User            @relation("ProviderReviews", fields: [providerId], references: [id])
+  rating                Int
+  comment               String?         
+}
+
+enum Role {
+  CLIENT
+  PROVIDER
+}
+
+enum AppointmentStatus {
+  BOOKED
```

Comment:
Given that providers need to confirm client appointment requests, how will you represent this state/status?